messagebus/src/vespa/messagebus/network/rpcsend.cpp

```diff
@@ -220,19 +220,19 @@ RPCSend::decode(vespalib::stringref protocolName, const vespalib::Version & vers
 void
 RPCSend::handleReply(Reply::UP reply)
 {
-    const IProtocol * protocol = _net->getOwner().getProtocol(reply->getProtocol());
-    if (!protocol || protocol->requireSequencing() || !_net->allowDispatchForEncode()) {
-        doHandleReply(protocol, std::move(reply));
+    if (!_net->allowDispatchForEncode()) {
+        doHandleReply(std::move(reply));
     } else {
-        auto rejected = _net->getExecutor().execute(makeLambdaTask([this, protocol, reply = std::move(reply)]() mutable {
-            doHandleReply(protocol, std::move(reply));
+        auto rejected = _net->getExecutor().execute(makeLambdaTask([this, reply = std::move(reply)]() mutable {
+            doHandleReply(std::move(reply));
         }));
         assert (!rejected);
     }
 }
 
 void
-RPCSend::doHandleReply(const IProtocol * protocol, Reply::UP reply) {
+RPCSend::doHandleReply(Reply::UP reply) {
+    const IProtocol * protocol = _net->getOwner().getProtocol(reply->getProtocol());
```

Comment:
Can this be `nullptr` or do we assume the protocol is always valid since we've already checked it as part of the initial request handling? Just wondering since the old code did a `if (!protocol` check (even though the check itself doesn't preclude the protocol pointer from potentially being dereferenced later in `doHandleReply`).